src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -682,19 +682,22 @@ public boolean isFipsStatusOk() {
     if (!isFips() || !isFipsSelfTestFailureSkipAbort()) {
       throw new UnsupportedOperationException("ACCP not built with FIPS_SELF_TEST_SKIP_ABORT");
     }
-    if (getSelfTestStatus() == SelfTestStatus.NOT_RUN) {
-      // If FIPS self tests haven't completed, give them a 5s timeout to complete.
+    // For "FipsStatus", we only care about AWS-LC's self test status.
+    SelfTestStatus status = SelfTestSuite.AWS_LC_SELF_TESTS.getCachedResult().getStatus();
+    if (status == SelfTestStatus.NOT_RUN) {
+      // If FIPS self tests haven't completed, give them a 3s timeout to complete.
       final long timeout = 3 * 1000;
       final long deadline = System.currentTimeMillis() + timeout;
-      while (getSelfTestStatus() == SelfTestStatus.NOT_RUN) {
+      while (status == SelfTestStatus.NOT_RUN) {
```

Comment:
Nit: You are checking `status == SelfTestStatus.NOT_RUN` twice in a row, once in the if statement and then immediately again in the while statement. This should be able to be refactored and merged so that it's only checked once.

Reply:
i'm intentionally guarding the call to `System.currentTimeMillis()` with the first check on L687, but i suppose i can change this to a do/while